devito/core/gpu.py

```diff
@@ -69,6 +69,7 @@ def _normalize_kwargs(cls, **kwargs):
         # Blocking
         o['blockinner'] = oo.pop('blockinner', True)
         o['blocklevels'] = oo.pop('blocklevels', cls.BLOCK_LEVELS)
+        o['skewing'] = oo.pop('skewing', False)
```

Comment:
that's a 1986 article, so I'm not buying that today there's any value in doing skewing+interchange only (especially with devito apps in mind), unless you now come back with (compelling) evidence

also, this would break our assumption that we see skewing as a subpass of blocking?

IMHO:

* if we see it as a subpass of blocking, then it makes sense to keep everything in blocking.py, and it also makes sense to call it blockskew -- just like all other blocking options it should start with "block". Here skewing would be the trigger to enable tiling along the skewing (not skewed) dimensions too?
* if it's not a subpass of blocking, then OK with calling it `skewing`, but then we should revisit the idea of keeping everything inside `blocking.py`. Maybe just use `misc.py` then.

Reply:
Well, I agree. That is why I said 'no perf gains == no value.'
So my opinion: 
- It should be a sub-pass of blocking, but it should also be tested for correctness as a standalone operation.
99% of the cases will be used in combination with blocking.